tests/ut/ir/transforms/test_derive_call_directions.py

```
@@ -0,0 +1,363 @@
+                x: pl.Tensor[[64], pl.FP32],
+                dst: pl.Tensor[[64], pl.FP32],
+            ) -> pl.Tensor[[64], pl.FP32]:
+                r: pl.Tensor[[64], pl.FP32] = self.kernel(x, dst)
+                return r
+
+        return _RewriteUserCall(call_dirs).run(Prog)
+
+    def test_input_with_output_rejected(self):
+        # Position 0 is callee In; using Output there must fail.
+        prog = self._build_program([ir.ArgDirection.Output, ir.ArgDirection.OutputExisting])
+        with pytest.raises(Exception):  # noqa: PT011
```

Ruff (0.15.10)

[warning] Do not assert blind exception: `Exception`

(B017)